vulnerability_report.md

```diff
@@ -0,0 +1,21 @@
+A1 injection:
+
+The file upload function in taskManager.views.py loads direct user input from the filename field into the (name) field in the function.  This allows us to make a SQL injection attack by adding a ' to our filename and adding an SQL query after it.  We did that and requested the admin password from auth_users.  When we clicked on the file we uploaded we got an error that showed the hashed password of the admin user.  This password is weak enough to be cracked if it was worth the trouble.
+
+We used an object relation manager to clean the data before allowing it to be passed on to the database.
+
+A2 crummy auth:
+
+The registration form in forms.py uses exclude rather than fields, and someone forgot to exclude some pretty important fields.  We added imput lines to the form by modifying the html and added superuser and staff permissions to our guy.  And then we made some more groups after accessing the admin panel.  Fix this by switching the excludes to a friendly whitelisted fields.
+
+A3 XSS
+
+The |safe method used to display / update the username in the site's forms accept javascript as a valid entry, and will execute when the name is displayed.  We changed our login name to an alert.  One could also change it to a redirect with hilarious/disturbing consequences.
+
+To fix: remove |safe from the forms that accept them.
+
+A4 Insecure Direct Object References
+
+There are tons of places in the code where one can directly access objects via the url without being properly authenticated.  For instance, the delete button on a user's task page: we logged in as user A and took note of a task that user was assigned.  In our case it was /taskManager/2/6.  We then logged in as user B and modified the HTML to direct the delete button away from the task it was assigned to and to /taskManager/2/task_delete/6.  When we clicked the button, we weren't able to see the redirected page, but the task was deleted - we confirmed by logging back in as user A.
+
+To fix: Add authentication to views that receive form input that ensures that a user can only perform actions related to objects they own.  Make sure object.user == request.user, etc.
```

Comment:
I'm not seeing this in any of the code in this pull request. What file did you modify?